### PR TITLE
add shouldComponentUpdate to blocks

### PR DESCRIPTION
**Pull request by nileshgulia1**
related #1694

### CODE REVIEW

File: src/components/manage/Blocks/Text/Edit.jsx [hunk 1]
```diff
@@ -123,6 +123,19 @@ class Edit extends Component {
...
+   * @param {*} nextProps
+   * @param {*} nextState
+   * @returns {boolean}
+   * @memberof Editq
+   */
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.state.addNewBlockOpened !== nextState.addNewBlockOpened) {
+      return true;
+    }
+    return this.props.data.text !== nextProps.data.text;
```
[the remark this answers is not in the document]
Comment (nileshgulia1) [reply src/components/manage/Blocks/Text/Edit.jsx:136]: Changed to using lodash `isEqual` method. Thanks for reporting :)

File: src/components/manage/Blocks/Text/Edit.jsx [hunk 2]
```diff
@@ -123,6 +123,19 @@ class Edit extends Component {
...
+   * @param {*} nextProps
+   * @param {*} nextState
+   * @returns {boolean}
+   * @memberof Edit
+   */
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.state.addNewBlockOpened !== nextState.addNewBlockOpened) {
+      return true;
+    }
+    return !isEqual(this.props.data.text, nextProps.data.text);
```
Comment (tiberiuichim) [inline src/components/manage/Blocks/Text/Edit.jsx:136]: Why not use `props.selected` here, too?

**Review by tiberiuichim — CHANGES_REQUESTED** [review]
In the interest of brevity, maybe we can change it like:

Instead of:
```
  shouldComponentUpdate(nextProps) {
    if (this.props.selected) {
      return true;
    }
    return !isEqual(this.props.data, nextProps.data);
  }
```

have:

```
  shouldComponentUpdate(nextProps) {
    return this.props.selected || !isEqual(this.props.data, nextProps.data);
  }
```

**Comment by nileshgulia1** [comment]
@tiberiuichim I have refactored some methods. You can have a look now :)

**Comment by tiberiuichim** [comment]
Nice! I'll test the code tomorrow

**Comment by tiberiuichim** [comment]
I'll test the code tomorrow.

**Review by tiberiuichim — APPROVED** [review]

**Comment by tisto** [comment]
todo:

- @tiberiuichim solves the merge conflict
- @sneridagh and @nzambello review it
- eau de web tests it in the alpha2

**Comment by tisto** [comment]
@nileshgulia1 @tiberiuichim @avoinea if one of you can resolve the merge conflict we can merge and release another alpha.

**Comment by nileshgulia1** [comment]
@tisto Done !

**Comment by nileshgulia1** [comment]
@plone/volto-team is there still any blocker for this PR, or something I could look into?

**Comment by tisto** [comment]
@plone/volto-team decision: this will go into 9.x.x. @tiberiuichim will create an issue regarding react.Memo.

**Comment by tiberiuichim** [comment]
@sneridagh Should be good to go, if cypress tests pass.

**Pull request merged**